src/contactmanagementsoftware/service/ChainOfResponsibility/CasualAcquaintanceValidator.java

```diff
@@ -0,0 +1,49 @@
+package contactmanagementsoftware.service.ChainOfResponsibility;
+
+import contactmanagementsoftware.entity.Contact;
+
+public class CasualAcquaintanceValidator extends ContactValidator {
+    private static final String CONTACT_TYPE = "class contactmanagementsoftware.CasualAcquaintance";
+
+    public CasualAcquaintanceValidator(ContactValidator contactValidator) {
+        super(contactValidator);
+    }
+
+    @Override
+    public boolean validateContact(Contact contact) {
+        if (contact.getClass().toString().equalsIgnoreCase(CONTACT_TYPE)) {
```

Comment:
actually `contact.getClass().getName()` will be evalutated as `CasualAcquaintance` I guess.
or better yet you can use `instanceof` operator

Reply:
same thought i am planning yo use instance of
but because now the CasualAcquaitances and others class is not impl as child of Contact

I will make the changes later when the composite DP is done